js/adapt-close.js

```diff
@@ -1,98 +1,75 @@
-define([
-  'core/js/adapt'
-], function(Adapt) {
-
-  const CloseView = Backbone.View.extend({
-
-    initialize: function() {
-      this.listenTo(Adapt, {
-        'navigation:closeButton': this.onCloseButton,
-        'close:confirm': this.onCloseConfirm,
-        'app:languageChanged': this.remove
-      }).render();
-    },
-
-    render: function() {
-      const data = this.model.toJSON();
-      data._globals = Adapt.course.get('_globals');
-
-      const template = Handlebars.templates.close;
-
-      this.setElement(template(data)).$el.appendTo('.nav__inner');
-    },
-
-    onCloseButton: function() {
-      const prompt = !Adapt.course.get('_isComplete') ?
-        this.model.get('_notifyPromptIfIncomplete') :
-        this.model.get('_notifyPromptIfComplete');
-
-      if (!prompt || !prompt._isEnabled) return Adapt.trigger('close:confirm');
-
-      Adapt.notify.prompt({
-        title: prompt.title,
-        body: prompt.body,
-        _prompts: [
-          {
-            promptText: prompt.confirm,
-            _callbackEvent: 'close:confirm'
-          },
-          {
-            promptText: prompt.cancel
-          }
-        ]
-      });
-    },
-
-    onCloseConfirm: function() {
-      // ensure that the browser prompt doesn't get triggered as well
-      const config = Adapt.course.get('_close');
-      config.browserPromptIfIncomplete = config.browserPromptIfComplete = false;
-
-      const scormWrapper = require('extensions/adapt-contrib-spoor/js/scorm/wrapper');
-      if (scormWrapper) {
-        const scormWrapperInstance = scormWrapper.getInstance();
-        if (scormWrapperInstance.lmsConnected && !scormWrapperInstance.finishCalled) {
-          scormWrapperInstance.finish();
-        }
-      }
-
-      if (config._button._closeViaLMSFinish) return;
-
-      top.window.close();
-    }
+import Backbone from 'backbone';
+import Adapt from 'core/js/adapt';
+import navigation from 'core/js/navigation';
+import NavigationButtonModel from 'core/js/models/NavigationButtonModel';
+import CloseNavigationButtonView from './CloseNavigationButtonView';
 
-  });
+class AdaptClose extends Backbone.Controller {
 
-  function onBeforeUnload(config) {
-    return !Adapt.course.get('_isComplete') ?
-      config.browserPromptIfIncomplete || undefined :
-      config.browserPromptIfComplete || undefined;
+  initialize() {
+    this.listenToOnce(Adapt, 'adapt:start', this.onStart);
+  }
+
+  get config () {
+    return Adapt.course.get('_close');
+  }
+
+  static get globalsConfig() {
+    return Adapt.course.get('_globals')?._extensions?._close;
+  }
+
+  onLanguageChange() {
+    $(window).off('beforeunload.close');
+    // have to wait until the navbar is ready
+    Adapt.once('navigationView:postRender', this.onStart);
   }
 
-  function initialise() {
+  onStart() {
     const config = Adapt.course.get('_close');

```

Comment:
```suggestion
    const config = Adapt.course.get('_close');
```
You have `this.config`.